frame/contracts/src/migration.rs

```diff
@@ -16,24 +16,20 @@
 // limitations under the License.
 
 use crate::{Config, Weight, Pallet};
-use frame_support::{
-	storage::migration,
-	traits::{GetPalletVersion, PalletVersion, PalletInfoAccess, Get},
-};
+use frame_support::{storage::migration, traits::{StorageVersion, PalletInfoAccess, Get}};
 
 pub fn migrate<T: Config>() -> Weight {
 	let mut weight: Weight = 0;
 
-	match <Pallet<T>>::storage_version() {
-		Some(version) if version == PalletVersion::new(3, 0, 0) => {
-			weight = weight.saturating_add(T::DbWeight::get().writes(1));
-			migration::remove_storage_prefix(
-				<Pallet<T>>::name().as_bytes(),
-				b"CurrentSchedule",
-				b"",
-			);
-		}
-		_ => (),
+	if StorageVersion::get::<Pallet<T>>() == 3 {
+		weight = weight.saturating_add(T::DbWeight::get().writes(1));
+		migration::remove_storage_prefix(
+			<Pallet<T>>::name().as_bytes(),
+			b"CurrentSchedule",
+			b"",
+		);
+
+		StorageVersion::new(4).put::<Pallet<T>>();
```

Comment:
Why do we need this line? Wont it be overwritten by the other storage version `put` in `migrate()`?

Reply:
As written in my message above, we don't write the storage version automatically anymore into the storage.

Reply:
While it feels like a step backwards, I agree that being explicit is better in this case, and it least for now.

Reply:
ambivalent, feels like a step backwards. Having it automatically written per `runtime_upgrade` is always what you want to do.

Basically you are allowing `pallet_storage_version` and `current_storage_version` to go out of sync. When is this useful?

Reply:
> Basically you are allowing `pallet_storage_version` and `current_storage_version` to go out of sync. When is this useful?

This is not useful. However, if you don't have executed the migration your fucked anyway.

IDK. I can bring the automatic update back. This way it is less magic. Especially useful for when we finally ban pallet `on_runtime_upgrade`, because we don't need to be afraid that the version is updated before the migration is executed in the runtime file.

Reply:
okay lets keep it manual for now then, less "magic".